src/robusta/runner/telemetry_service.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+from enum import Enum
+from time import sleep
+import sentry_sdk
+import requests
+import threading
+from src.robusta.model.config import Registry, Telemetry
+from hikaru.model import NodeList
+
+class TelemetryLevel(Enum):
+    NONE = 0,
+    USAGE = 1,
+    ERROR = 2 
+
+class TelemetryService: 
+    def __init__(
+        self,
+        telemetry_level: TelemetryLevel,
+        endpoint: str,
+        periodic_time_sec: float,
+        registry : Registry
+    ):
+        self.telemetry_level = telemetry_level
+        self.endpoint = endpoint
+        self.registry = registry
+        self.periodic_time_sec = periodic_time_sec
+
+        sentry_dsn = os.environ.get("SENTRY_DSN", "")
+        if self.telemetry_level == TelemetryLevel.ERROR and sentry_dsn:
+            logging.info(f"Telemetry set to include error info, Thank you for helping us improve Robusta.")
+            try:
+                sentry_sdk.init(sentry_dsn, traces_sample_rate=float(os.environ.get("SENTRY_TRACES_SAMPLE_RATE", 0.5)))
+            except Exception as e:
+                logging.error(f"Sentry error: {e}")
+                pass
+
+        self.__thread = threading.Thread(target=self.__log_periodic)
+        self.__thread.start()
+
+
+    def __log_periodic(self):
+
+        while(True):
+            tele = self.registry.get_telemetry()
+          
+            current_nodes: NodeList = NodeList.listNode().obj
```

Comment:
If an exception is thrown here, the telemetry thread will stop.
I would surround this logic with try/except instead of the one inside the `log` method